app/(moderation)/dashboard/profile/[id]/page.tsx

```diff
@@ -16,12 +17,14 @@ export default async function ModerationUserProfile({
   params: { id: string }
 }) {
   const profile = await getProfileByUserId(params.id)
+  const user = profile && (await findUserByEmail(profile.userEmail))
 
-  if (!profile || !requireUserRoles([Role.MODERATOR])) redirect(AppRoutes.home)
+  if (!profile || !user || !requireUserRoles([Role.MODERATOR]))
+    redirect(AppRoutes.home)
 
   return (
     <div className={styles.wrapper}>
-      <ModerationActionHeader userProfile={profile} />
+      <ModerationActionHeader userProfile={profile} userRoles={user.roles} />
```

Comment:
**nitpick:** I think you don't have to pass user roles because you can pull it out straight in component using `useSession` hook

Reply:
The user we're working on dashboard/profile:id is not the one in the session. 
In this case it would be the one that owns the profile, hence useSession wouldn't work (cuz it works with moderator that moderates user profiles). 
I could get roles using existent prop (userProfile) but it would mean redundant call in order to obtain these roles once again. 
If such approach would prove to be more beneficial, then we'll keep next commit that I'll provide in few minutes from this comment.

Reply:
Okay, nevermind as fas as I can see i'd have to create container server component with call to prisma db to obtain these roles once again only to pass data as children/props down to the client component so it wouldn't make sense I suppose?